src/emailagent.cpp

```diff
@@ -1444,8 +1444,7 @@ bool EmailAgent::saveAttachmentToDownloads(const QMailMessageId &messageId, cons
     const QMailMessage message (messageId);
     const QMailMessagePart::Location location(attachmentLocation);
     QMailAccountId accountId = message.parentAccountId();
-    QString attachmentDownloadFolder = QStandardPaths::writableLocation(QStandardPaths::DownloadLocation) + "/mail_attachments/"
-            + QString::number(accountId.toULongLong()) +  "/" + attachmentLocation;
+    QString attachmentDownloadFolder = QStandardPaths::writableLocation(QStandardPaths::DownloadLocation);
```

Comment:
I really like this improvement (because now it would be possible to actually find attachments with a file browser).
But this code looks a lot like an existing file by the same name as a attachment being saved will count as if it is already downloaded. (e.g. if you receive multiple emails with "ticket.pdf", only the first one gets downloaded, and subsequent emails will appear as if they contained the first instance of the file)

Reply:
Very good point.. some form of isolation between individual emails, or some form of sha1sum check to determine whether the previously downloaded attachment file with the same name is in fact the same attachment, would be required.

Reply:
Yes, this is an important question. Below, in general comments, I will describe our vision of the problem.